AutomatedTesting/Gem/PythonTests/scripting/FileMenu_Default_NewAndOpen.py

```diff
@@ -50,38 +51,43 @@ class TestFileMenuDefaultNewOpen:
 
     @pyside_utils.wrap_async
     async def run_test(self):
-        # 1) Open Script Canvas window (Tools > Script Canvas)
+
+        # Preconditions
+        general.idle_enable(True)
+
+        # 1) Open Script Canvas window and wait for it to render(Tools > Script Canvas)
         general.open_pane("Script Canvas")
+        helper.wait_for_condition(lambda: general.is_pane_visible("Script Canvas"), TIME_TO_WAIT)
 
         # 2) Get the SC window object
         editor_window = pyside_utils.get_editor_main_window()
         sc = editor_window.findChild(QtWidgets.QDockWidget, "Script Canvas")
         sc_main = sc.findChild(QtWidgets.QMainWindow)
         sc_tabs = sc_main.findChild(QtWidgets.QTabWidget, "ScriptCanvasTabs")
 
-        # wait for the intial tab count
-        general.idle_wait(GENERAL_WAIT)
+        # 3) Get the initial tab count and verify it's zero. Save the tab count for verification in step 5
+        result = helper.wait_for_condition(lambda: sc_tabs.count() == 0, TIME_TO_WAIT)
+        Report.info(f"{Tests.initialTabTestResults}: {result}")
         initial_tabs_count = sc_tabs.count()
 
-        # 3) Trigger File->New action
+        # 4) Trigger open new graph file action
         action = pyside_utils.find_child_by_pattern(
-            sc_main, {"objectName": "action_New_Script", "type": QtWidgets.QAction}
-        )
+            sc_main, {"objectName": "action_New_Script", "type": QtWidgets.QAction})
         action.trigger()
 
-        # 4) Verify if New tab is opened
-        general.idle_wait(GENERAL_WAIT)
-        Report.info(f"{Tests.new_action}: {sc_tabs.count() == initial_tabs_count + 1}")
+        # 5) Check tab count again to verify a new tab has been opened
+        result = helper.wait_for_condition(lambda: sc_tabs.count() == initial_tabs_count + 1, TIME_TO_WAIT)
+        Report.info(f"{Tests.newTabTestResults}: {result}")
 
-        # 5) Trigger File->Open action
+        # 6) Trigger open file popup action then close the popup.
         action = pyside_utils.find_child_by_pattern(sc_main, {"objectName": "action_Open", "type": QtWidgets.QAction})
         pyside_utils.trigger_action_async(action)
-        general.idle_wait(GENERAL_WAIT)
         popup = await pyside_utils.wait_for_modal_widget()
-        Report.info(f"{Tests.open_action}: {popup and 'Open' in popup.windowTitle()}")
+        result = popup and 'Open' in popup.windowTitle()
+        Report.info(f"{Tests.fileMenuTestResults}: {result}")
         popup.close()
 
-        # 6) Close Script Canvas window
+        # 7) Close Script Canvas window
         general.close_pane("Script Canvas")
```

Comment:
Since this is used more than once or twice, you should store this into a variable so you don't get syntax errors and it's easier to update references in the event of a window title change.

Reply:
added a constant for the "script canvas" string we use in several lines. Also, updated the steps of the test outlined at the top of the file to match the new flow and formatting.

Reply:
Now using const "SCRIPT_CANVAS_PANE"